dashboard/src/actions/overviewActions.js

```diff
@@ -405,3 +403,10 @@ export const getEditedMetadata =
     }
     dispatch(updateDataset(dataset, editedMetadata));
   };
+
+const markEditableFields = (item, isDirty) => {
+  item[CONSTANTS.IS_DIRTY_NAME] = isDirty;
+  item[CONSTANTS.IS_DIRTY_SERVER_DELETE] = isDirty;
+
+  return item;
+};
```

Comment:
I think I remember seeing code like this before somewhere, and commenting that this seems odd. It makes sense to have a common place to *clear* all "dirty" flags at once before displaying, but I'm not sure it's possible to have a situation where both are *set* at the same time since I don't remember the UI having a mode where you can freely edit both and then submit both changes at once. In any case, I only see this being called with `isDirty` set to `false`; in which case it seems that `isDirty` is unnecessary, and `clearEditableFields` might be a better name?

Reply:
Yes, it got removed on code merge. So, re-written it. Modified the function name.